src/utils.ts

```diff
@@ -16,17 +16,45 @@ export class SdkError extends Error {
 
   // ignoring this due to a coverage issue relating to babel. https://github.com/Microsoft/TypeScript/issues/13029
   /* istanbul ignore next */
-  constructor (errorType: SdkErrorTypes | null, message: string | null, details?: any) {
-    super(message);
+  constructor (errorType: SdkErrorTypes | null, messageOrError: string | Error, details?: any) {
+    /* if a Error is passed in, use its message and name properties */
+    const isError = messageOrError && messageOrError instanceof Error;
+    super(isError ? (messageOrError as any).message : messageOrError);
+
+    if (isError) {
+      this.name = (messageOrError as any).name;
+    }
+
     this.type = errorType || SdkErrorTypes.generic;
     this.details = details;
   }
 }
 
-export const throwSdkError = function (this: GenesysCloudWebrtcSdk, errorType: SdkErrorTypes | null, message: string | null, details?: any): void {
-  const error = new SdkError(errorType, message, details);
+/**
+ * This will create an `SdkError`, emit the error on `sdk.on('sdkError', error)`,
+ *  and return the error. It will not `throw` the error. It is up to the caller 
+ *  on what to do with it. 
+ * @param this sdk instance
+ * @param errorType SdkError type
+ * @param message message as string or Error instance
+ * @param details any additional details to log with the error
+ */
+export const createAndEmitSdkError = function (this: GenesysCloudWebrtcSdk, errorType: SdkErrorTypes | null, messageOrError?: string | Error, details?: any): SdkError {
+  const error = new SdkError(errorType, messageOrError, details);
   this.emit('sdkError', error);
-  throw error;
+  return error;
+};
+
+/**
+ * This will create an `SdkError`, emit the error on `sdk.on('sdkError', error)`,
+ *  and `throw` the error. 
+ * @param this sdk instance
+ * @param errorType SdkError type
+ * @param message message as string or Error instance
+ * @param details any additional details to log with the error
+ */
+export const throwSdkError = function (this: GenesysCloudWebrtcSdk, errorType: SdkErrorTypes | null, messageOrError?: string | Error, details?: any): void {
```

Comment:
kind of weird that this is just a pass-through fn. is there really a point to it? Can we use either this fn or the one above everywhere?

Reply:
Yeah, let's just use `createAndEmit..` since it isn't super clear that this function throws the error. Then the consumer can decided if it wants to throw it or something else